pkg/ebpfsyncer/ebpfsyncer.go

```diff
@@ -80,93 +84,176 @@ func (e *ebpfSingleton) SyncInterfaceIngressRules(
 		}()
 	}
 
+	// Create a new manager if none exists.
+	if err := e.createNewManager(); err != nil {
+		return err
+	}
+
 	// For delete operations, detach all interfaces and run a cleanup, set managed interfaces and the
 	// manager to empty / nil values, then return.
-	// TODO: This should purge all XDP attachments on all interfaces even if we are not managing them.
-	// Alternatively, we need something like method XDPInterfacePurge() that would purse all unmanaged interfaces
-	// both from the rules and from the interface attachements.
 	if isDelete {
-		logger.Info("Running detach and eBPF cleanup operation")
-		for intf := range e.managedInterfaces {
-			_, err := e.c.IngressNodeFwAttach([]string{intf}, true)
-			if err != nil {
-				return err
-			}
-		}
-		e.managedInterfaces = make(map[string]struct{})
-		err := e.c.CleaneBPFObjs()
-		if err != nil {
-			return err
-		}
-		e.c = nil
+		return e.resetAll()
```

Comment:
isDelete can be set if we just deleting one rules obj but not all objs right ? so we can't resetAll()

Reply:
With this PR, `isDelete` is only called when the `IngressNodeFirewallNodeState` object is deleted for that node, so this is a global delete operation when the reconciler reports that the object gets a delete.

For object creation or updates:
All rules are applied in an idempotent way, meaning that we go through all rules and diff what should be there vs what's there and deletion of individual rules is achieved that way.

Reply:
~~~
   80 // IngressNodeFwRulesLoader adds/updates/deletes ingress node firewall rules to the eBPF LPM MAP in an idempoten      t way.  
   81 // IngressNodeFwRulesLoader executes the following actions in order:
   82 // i)   Get eBPF objs to create/update eBPF maps and get map info.
   83 // ii)  Build a map of valid ebpfKeys pointing to the ebpfRules that should be associated to them (built from
   84 //      ifaceIngressRules).
   85 // iii) Get stale keys (= keys inside the eBPF map but not inside the currently desired ruleset).
   86 // iv)  Purge all stale keys from the eBPF map.
   87 // v)   Add/update all keys. This is an idempotent action and non-existing keys are added whereas existing keys
   88 //      are updated.
   89 // vi)  Generate ingress node firewall events.
   90 // In the context of this method, stale keys are keys that figure inside the eBPF map but that are not generated
   91 // during step ii) from the provided ingressRules slice.
   92 func (infc *IngNodeFwController) IngressNodeFwRulesLoader(
   93     ifaceIngressRules map[string][]v1alpha1.IngressNodeFirewallRules) error {
~~~